bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -178,9 +261,39 @@ public void propertyChange(PropertyChangeEvent evt) {
 			}
 
 			// Sky brightness warnings
+
 			Double sb = a.getMin(Circumstance.TOTAL_SKY_BRIGHTNESS, false);
-			if (sb != null && !a.getObs().getConditions().containsSkyBrightness(sb))
+
+			final Option<Interval> skyBackgroundInterval = night.flatMap(n -> {
+				final Obs obs = a.getObs();
+				final SkyBackgroundSolver skyBackgroundSolver = new SkyBackgroundSolver(site, obs::getCoords, sb);
+				final SortedSet<Interval> intervals = skyBackgroundSolver.solve(n).getIntervals();
+				if (intervals.isEmpty()) {
+					return None.instance();
+				} else {
+					final Interval interval = intervals.first();
+					// For some reason the default background interval is larger than the night interval
+					if (interval.getStart() >= n.getStart() && interval.getEnd() <= n.getEnd()) {
+						return None.instance();
+					}
+					return new Some<>(intervals.first());
```

Comment:
My understanding was that the solver would clip the intervals based on what is passed to `solve`, but that's not happening?  At any rate, won't it often be the case that the interval will fall inside of the night?  That is, `interval.getStart() >= n.getStart() && interval.getEnd() <= n.getEnd()` seems like the _usual_ case?

Reply:
Yes it seems like the solver should always clip the intervals. Why are we only looking at the first such interval? If the moon rises and sets there might be two usable intervals, seems like.

Calculating this stuff is expensive and it's already been done at least once in the `Variant` to determine whether the obs is schedulable in the first place, so it seems like we could cache that computation and rely on it rather than recomputing here. This is fine for now but this class is getting messy and needs some cleanup eventually.

Reply:
By "drop it all together", what exactly do you mean? Omit it from 2018B due to the complexity at this point, or just go with the flawed result?

Reply:
@sraaphorst I meant dropping just the sky background check, but I'll try to fix it.